snapcraft/internal/meta/snap.py

```diff
@@ -259,6 +281,15 @@ def to_dict(self):  # noqa: C901
                 snap_dict[key] = dict()
                 for name, slot in sorted(self.slots.items()):
                     snap_dict[key][name] = slot.to_dict()
+            elif key == "grade":
+                if self.grade is None:
```

Comment:
I think it'd be cleanest if you took this section and just chunked it into a `@property` for `grade`.  Then there wouldn't be any special casing in `to_dict()`.  And it would just return the required grade if unset.

Reply:
You're right, that's gonna break the __dict__... pass on this

Reply:
I don't think this is doable with the current solution, I am planning on changing this dict dance a bit, let's see how that plays out